### ISSUE TITLE
Support to use `tsconfig-paths` in TypeScript wdio config file

### ISSUE THREAD

**Issue by jkostrhun** [webdriverio/webdriverio#6500]
**Environment (please complete the following information):**
 - **WebdriverIO version:** [7.0.7]
 - **Mode:** [WDIO Testrunner]
 - **If WDIO Testrunner, running sync/async:** [sync]
 - **Node.js version:** [14.15.5]
 - **NPM version:** [yarn 1.22.10]
 - **Platform name and version:** [macOS Mojave]

**Config of WebdriverIO**
filename: wdio.conf.ts

```
import module1 from 'modules/module/module1'

const config: WebdriverIO.Config = {
    runner: 'local',
    specs: [
        './specs/**/*.ts'
    ],
    capabilities: [{
        browserName: 'chrome',
    }],
    framework: 'mocha',
    reporters: ['spec'],
    mochaOpts: {
        ui: 'bdd',
    },
    autoCompileOpts: {
        autoCompile: true,
        tsNodeOpts: {
            transpileOnly: true,
            project: 'tsconfig.json',
            require: ['tsconfig-paths'] // tried also ['tsconfig-paths/register']
        }
    },
}
export {config}
```

**Describe the bug**
When I try to import some module described in paths in tsconfig.json in scope of WDIO TypeScript config file using 'tsconfig-paths' I'm getting Error: Cannot find module 'modules/module/module1'

**To Reproduce**
Have following in tsconfig.json
```
    "baseUrl": "./",
    "paths": {
      "modules/*": ["modules/*"],
    }
```

Some module in specified path.

Run `wdio run wdio.conf.ts`

**Expected behavior**
WDIO will find module, like it's working now with following solution:

file: wdio.conf.js

```
require('ts-node').register({ transpileOnly: true })
require('tsconfig-paths').register()
module.exports = require('./wdio.conf.ts')
```

file: wdio.conf.ts
```
import module1 from 'modules/module/module1'

const config: WebdriverIO.Config = {
    runner: 'local',
    specs: [
        './specs/**/*.ts'
    ],
    capabilities: [{
        browserName: 'chrome',
    }],
    framework: 'mocha',
    reporters: ['spec'],
    mochaOpts: {
        ui: 'bdd',
    },
}
export {config}
```

Run `wdio run wdio.conf.js`

**Log**

Error: Cannot find module 'modules/module/module1'
Require stack:
- /Users/jkostrhun/wdio.conf.ts
- /Users/jkostrhun/node_modules/@wdio/config/build/lib/FileSystemPathService.js
- /Users/jkostrhun/node_modules/@wdio/config/build/lib/ConfigParser.js
- /Users/jkostrhun/node_modules/@wdio/config/build/index.js
- /Users/jkostrhun/node_modules/@wdio/cli/build/launcher.js
- /Users/jkostrhun/node_modules/@wdio/cli/build/index.js
- /Users/jkostrhun/node_modules/@wdio/cli/bin/wdio.js
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:880:15)
    at Function.Module._load (internal/modules/cjs/loader.js:725:27)
    at Module.require (internal/modules/cjs/loader.js:952:19)
    at require (internal/modules/cjs/helpers.js:88:18)
    at Object.<anonymous> (/Users/jkostrhun/wdio.conf.ts:1:1)
    at Module._compile (internal/modules/cjs/loader.js:1063:30)
    at Module.m._compile (/Users/jkostrhun/node_modules/ts-node/src/index.ts:1056:23)
    at Module._extensions..js (internal/modules/cjs/loader.js:1092:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/jkostrhun/node_modules/ts-node/src/index.ts:1059:12)
    at Module.load (internal/modules/cjs/loader.js:928:32) {
  code: 'MODULE_NOT_FOUND',
  requireStack: [
    '/Users/jkostrhun/wdio.conf.ts',
    '/Users/jkostrhun/node_modules/@wdio/config/build/lib/FileSystemPathService.js',
    '/Users/jkostrhun/node_modules/@wdio/config/build/lib/ConfigParser.js',
    '/Users/jkostrhun/node_modules/@wdio/config/build/index.js',
    '/Users/jkostrhun/node_modules/@wdio/cli/build/launcher.js',
    '/Users/jkostrhun/node_modules/@wdio/cli/build/index.js',
    '/Users/jkostrhun/node_modules/@wdio/cli/bin/wdio.js'
  ]
}

**Comment by ipopa** [comment]
```
mochaOpts: {
       requires: ['tsconfig-paths/register'],
        ...
    }

```

**Comment by jkostrhun** [comment]
@ipopa I tried it, but this allows you to use 'tsconfig-paths' in .spec files, not in wdio config file itself

**Comment by christian-bromann** [comment]
I suggest to add another option to `autoCompileOpts` named `tsConfigPathOpts` and if existing we also require `tsconfig-paths` within the [compiler function](https://github.com/webdriverio/webdriverio/blob/main/packages/wdio-config/src/utils.ts#L106-L115). Contributions welcome!

**Comment by christian-bromann** [comment]
Implemented in #6581
